workers/unity/Assets/Fps/Scripts/Config/MapBuilder.cs

```diff
@@ -68,21 +132,119 @@ private void PlaceTiles()
+        }
+
+        Debug.LogError($"Failed to load resource at {resourcePath}");
+        return false;
+    }
+
+    private Transform MakeGroup(string groupName)
+    {
+        var group = new GameObject(groupName).transform;
+        group.parent = transform;
+        group.localPosition = Vector3.zero;
+        group.localRotation = Quaternion.identity;
+        group.localScale = Vector3.one;
+        return group;
+    }
+
+    private void FillSurround()
+    {
+        for (var i = -groundLayers; i < groundLayers; i++)
+        {
+            var offset = TileSeparation * 4 * i + TileSeparation * 2;
+            MakeEdge(offset, 0);
+            MakeEdge(offset, 90);
+            MakeEdge(offset, 180);
+            MakeEdge(offset, 270);
+        }
+
+        for (var i = 0; i < 360; i += 90)
+        {
+            MakeCorner(i);
+        }
+    }
+
+    private void MakeEdge(float offset, int angle)
+    {
+        var rotation = Quaternion.Euler(0, angle, 0);
+
+        var floor = Instantiate(groundEdge,
+            rotation * new Vector3(offset, 0, groundLayers * TileSeparation * 4 + BlockSize * 0.25f),
+            rotation * Quaternion.Euler(90, 0, 0),
+            groundParentTransform);
+        floor.transform.localScale = new Vector3(144, 2, 1);
+
+        var wall = Instantiate(surroundWall,
+            rotation * new Vector3(offset, BlockSize * .5f, groundLayers * TileSeparation * 4 + BlockSize * .5f),
+            rotation,
+            surroundParentTransform);
+        wall.transform.localScale = new Vector3(144, 4, 1);
+
+        var wallFloor = Instantiate(groundEdge,
+            rotation * new Vector3(offset, BlockSize, groundLayers * TileSeparation * 4 + BlockSize),
+            rotation * Quaternion.Euler(90, 0, 0),
+            surroundParentTransform);
+        wallFloor.transform.localScale = new Vector3(144, 4, 1);
+
+        // Collision
+        var collisionHeight = 16;
+        var collision = Instantiate(surroundWall,
+            rotation * new Vector3(offset, BlockSize + collisionHeight * .5f,
+                groundLayers * TileSeparation * 4 + BlockSize * .5f),
+            rotation,
+            surroundParentTransform);
+        collision.transform.localScale = new Vector3(144 + 4, collisionHeight, 1); // Collisions overlap to fill corners
+        collision.gameObject.name = "Collision";
+
+        if (Application.isPlaying)
+        {
+            Destroy(collision.GetComponent<MeshRenderer>());
+            Destroy(collision.GetComponent<MeshFilter>());
+        }
+        else
+        {
+            DestroyImmediate(collision.GetComponent<MeshRenderer>());
+            DestroyImmediate(collision.GetComponent<MeshFilter>());
+        }
+    }
+
+    private void MakeCorner(int angle)
+    {
+        var rotation = Quaternion.Euler(0, angle, 0);
+        Instantiate(cornerPiece,
+            rotation * new Vector3(-groundLayers * TileSeparation * 4, 0, groundLayers * TileSeparation * 4),
```

Comment:
It could be, but it's only called four times and only in editor time so calculating it each time is pretty inconsequential.

What I could do to improve readability is make a property with a descriptive name like CornerOffset? Like,

`Vector3 CornerOffset => new Vector3(-groundLayers * TileSeparation * 4, 0, groundLayers * TileSeparation * 4)`

Reply:
in hindsight it felt like I was just bloating the class variables, so have done as you suggested and modified MakeCorner() to take in the offset as an argument